gax/src/main/java/com/google/api/gax/rpc/StubSettings.java

```diff
@@ -293,9 +307,31 @@ protected B self() {
      * call logic (such as retries and long-running operations), and also to pass to the transport
      * settings if an executor is needed for the transport and it doesn't have its own executor
      * provider.
+     *
+     * @deprecated Please use {@link #setBackgroundExecutorProvider(ExecutorProvider)} for setting
+     *     executor to use for running scheduled API call logic. To set executor for {@link
+     *     TransportChannelProvider}, please use {@link
+     *     TransportChannelProvider#withExecutor(Executor)} instead.
      */
+    @Deprecated
     public B setExecutorProvider(ExecutorProvider executorProvider) {
+      // For backward compatibility, this will set both executorProvider and
```

Comment:
Since `executorProvider` field is private and nobody knows about it except this class, can we instead of trying to keep the `executorProvider` and `backgroundExecutorProvider` in sync, just delete `executorProvider` and replace it with `backgroundExecutorProvider` everywhere (i.e. basically making `executorProvider` and `backgroundExecutorProvider` synonyms of the same thing, istead of being copies of the same thing)?

Reply:
(Explained in the comment above):

backgroundExecutorProvider is default to InstantiatingExecutorProvider (in ClientContext: https://github.com/googleapis/gax-java/pull/1355/files#diff-1471e90108f24262d8e1b215bdd0fdff64d159d2201ba72bda5df6dba58fdc7cR262), and executorProvider is now default to null.

In the default case, when both StubSettings#setExecutor and TrasnportChannelProvider#withExecutor are not set, we want to use default grpc executor for transportChannelProvider, so we need to keep executorProvider to be null for the default case (the check in ClientContext#create https://github.com/googleapis/gax-java/pull/1355/files#diff-6138cc9cd4c9af4ec5aceebe7a723585298a068fff7a7848946c452778cca794R186).